client-core/src/service/wallet_service.rs

```diff
@@ -23,16 +23,19 @@ pub struct Wallet {
     pub staking_keys: BTreeSet<PublicKey>,
     /// root hashes of multi-sig transfer addresses
     pub root_hashes: BTreeSet<H256>,
+    /// is this wallet watch only
+    pub watch_only: bool,
```

Comment:
I think `WalletRole` is a name too broad for this. Actually, I'm thinking of eliminating this boolean variable, because it doesn't really make much difference. A watch-only wallet is actually a fully functioning `BasicWallet` since it already has view key pair.

Reply:
Changed `restoreWatch` to `restoreBasic`, removed the `watch_only` field in `Wallet`.